app/src/main/java/nerd/tuxmobil/fahrplan/congress/base/SessionsAdapter.kt

```diff
@@ -0,0 +1,232 @@
+                        rowView.requireViewByIdCompat(R.id.session_list_item_no_video_view),
+                        rowView.requireViewByIdCompat(R.id.session_list_item_without_video_recording_view),
+                    )
+                    rowView.tag = viewHolder
+                }
+                TYPE_SEPARATOR -> {
+                    rowView = localInflater.inflate(R.layout.session_list_separator, parent, false)
+                    viewHolderSeparator = ViewHolderSeparator()
+                    viewHolderSeparator.text = rowView.requireViewByIdCompat(R.id.session_list_separator_title_view)
+                    rowView.tag = viewHolderSeparator
+                }
+                else -> {
+                    throw IllegalStateException("Unknown type: $type")
+                }
+            }
+        } else {
+            rowView = convertView
+            when (type) {
+                TYPE_ITEM -> {
+                    viewHolder = rowView.tag as ViewHolder
+                }
+                TYPE_SEPARATOR -> {
+                    viewHolderSeparator = rowView.tag as ViewHolderSeparator
+                }
+                else -> {
+                    throw IllegalStateException("Unknown type: $type")
+                }
+            }
+        }
+        when (type) {
+            TYPE_ITEM -> {
+                setItemContent(position, viewHolder!!)
+            }
+            TYPE_SEPARATOR -> {
+                setSeparatorContent(position, viewHolderSeparator)
+            }
+            else -> {
+                throw IllegalStateException("Unknown type: $type")
+            }
+        }
+        return rowView
+    }
+
+    fun getSession(position: Int): Session = list!![mMapper!![position]]
+
+    protected open fun resetItemStyles(viewHolder: ViewHolder) {
+        resetTextStyle(viewHolder.title, R.style.ScheduleListPrimary)
+        resetTextStyle(viewHolder.subtitle, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.speakers, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.lang, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.day, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.time, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.room, R.style.ScheduleListSecondary)
+        resetTextStyle(viewHolder.duration, R.style.ScheduleListSecondary)
+        viewHolder.withoutVideoRecording.setImageResource(R.drawable.ic_without_video_recording)
+    }
+
+    protected open fun resetTextStyle(textView: TextView, style: Int) {
+        textView.setTextAppearance(context, style)
+    }
+
+    protected abstract fun setItemContent(position: Int, viewHolder: ViewHolder)
+
+    private fun setSeparatorContent(position: Int, viewHolderSeparator: ViewHolderSeparator?) {
+        viewHolderSeparator!!.text!!.text = mSeparatorStrings!![mMapper!![position]]
+    }
+
+    override fun getViewTypeCount(): Int = NUM_VIEW_TYPES
```

Comment:
The return type can be omitted here and below.

Reply:
Should I also do this in Line 116?

Reply:
If it fits there..

Reply:
It's the `getSession` function, at the moment that's line 122